src/protocol/attribution/credit_capping.rs

```diff
@@ -196,6 +210,36 @@ where
     .await
 }
 
+async fn report_level_capping<F, C, T>(
```

Comment:
I would really appreciate a comment that explains how this fixes the root issue we are concerned about. I don't understand it right now.

Reply:
This comment is good. I think it would probably be better above:

```
pub async fn credit_capping<F, C, T>(
```